src/Microsoft.Diagnostics.Runtime/src/DbgEng/DebugSystemObjects.cs

```diff
@@ -29,41 +29,45 @@ public uint GetProcessId()
             InitDelegate(ref _getProcessId, VTable.GetCurrentProcessSystemId);
 
             using IDisposable holder = Enter();
-            int hr = _getProcessId(Self, out uint id);
-            return hr == S_OK ? id : 0;
+            HResult hr = _getProcessId(Self, out uint id);
+            return hr ? id : 0;
         }
 
-        private void SetCurrentSystemId(int id)
+        private HResult SetCurrentSystemId(int id)
         {
             InitDelegate(ref _setSystemId, VTable.SetCurrentSystemId);
 
-            int hr = _setSystemId(Self, id);
-            DebugOnly.Assert(hr == S_OK);
+            HResult hr = _setSystemId(Self, id);
+            DebugOnly.Assert(hr);
+            return hr;
         }
 
-        public void SetCurrentThread(uint id)
+        public HResult SetCurrentThread(uint id)
         {
             InitDelegate(ref _setCurrentThread, VTable.SetCurrentThreadId);
 
             using IDisposable holder = Enter();
-            int hr = _setCurrentThread(Self, id);
-            DebugOnly.Assert(hr == S_OK);
+            HResult hr = _setCurrentThread(Self, id);
+            DebugOnly.Assert(hr);
+            return hr;
         }
 
         public int GetNumberThreads()
         {
             InitDelegate(ref _getNumberThreads, VTable.GetNumberThreads);
 
             using IDisposable holder = Enter();
-            int hr = _getNumberThreads(Self, out int count);
-            DebugOnly.Assert(hr == S_OK);
+            HResult hr = _getNumberThreads(Self, out int count);
+            DebugOnly.Assert(hr);
             return count;
         }
 
         internal void Init()
         {
             InitDelegate(ref _getSystemId, VTable.GetCurrentSystemId);
-            _systemId = _getSystemId(Self, out int id);
```

Comment:
@NextTurn Actually there was one bug "fixed" with this checkin, it's this line.  We originally set the _systemId to be the HRESULT of the return of `IDebugSystemObjects::GetSystemId`, which is a bug.

I fixed that here, but I'm seeing odd behavior in tests, so I'm debugging the problem now.